src/EditorFeatures/Test2/FindReferences/FindReferencesTests.NamedTypeSymbols.vb

```diff
@@ -2311,6 +2311,174 @@ public class D { }
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Category", "RuleId", Scope = "member", Target = "~M:N.C1.[|C2|].M")]
+
+namespace N
+{
+    class C1
+    {
+        private class {|Definition:$$C2|}
+        {
+            void M()
+            {
+            }
+        }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem(44288, "https://github.com/dotnet/roslyn/issues/44288")>
+        <WpfTheory, CombinatorialData, Trait(Traits.Feature, Traits.Features.FindReferences)>
+        Public Async Function TestTypeReferenceInGlobalSuppression_GenericType(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document><![CDATA[
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Category", "RuleId", Scope = "member", Target = "~T:N.[|C|]`1")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Category", "RuleId", Scope = "member", Target = "~M:N.[|C|]`1.M")]
+
+namespace N
+{
+    class {|Definition:$$C|}<T>
+    {
+        void M()
+        {
+        }
+    }
+}]]>
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem(44288, "https://github.com/dotnet/roslyn/issues/44288")>
+        <WorkItem(44401, "https://github.com/dotnet/roslyn/issues/44401")>
+        <WpfTheory(Skip:="https://github.com/dotnet/roslyn/issues/44401")>
+        <CombinatorialData, Trait(Traits.Feature, Traits.Features.FindReferences)>
+        Public Async Function TestTypeReferenceInGlobalSuppressionParameter(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Category", "RuleId", Scope = "member", Target = "~M:N.D.M(N.[|C|])")]
+
+namespace N
+{
+    class D
+    {
+        void M([|C|] c)
+        {
+        }
+    }
+
+    class {|Definition:$$C|}
+    {
+    }
+}
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem(44288, "https://github.com/dotnet/roslyn/issues/44288")>
+        <WorkItem(44401, "https://github.com/dotnet/roslyn/issues/44401")>
+        <WpfTheory(Skip:="https://github.com/dotnet/roslyn/issues/44401")>
+        <CombinatorialData, Trait(Traits.Feature, Traits.Features.FindReferences)>
+        Public Async Function TestTypeReferenceInGlobalSuppressionParameter_GenericType(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document><![CDATA[
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Category", "RuleId", Scope = "member", Target = "~M:N.D.M(N.[|C|]{System.Int32})")]
```

Comment:
what about a named type referenced in a generic type arg list.  i.e. `X{[|C|]}`

Reply:
Good point, I have added it to https://github.com/dotnet/roslyn/issues/44401. I plan to address it in the next PR.